### DISCUSSION TITLE
Welcome to EleksTubeIPS Discussions

### DISCUSSION THREAD

**Discussion by judge2005** [judge2005/EleksTubeIPS#2] [Announcements]
## 👋 Welcome!
Use this space to share ideas and feedback for future development. I have several things cued up if I find time.

**Comment by IPSude** [comment]
Hi, I'm loving the idea of custom firmware for the Eleksmaker IPS clock. 
What lead me here was your weatherforecast functionality since I can't seem to get the basic weather to work, it always display the same thing 0°...
I have the gen 2 version, do you have any plans of making a gen 2 compatible version of your firmware?

**Reply by judge2005** [reply]
I do build one, but AFAIK you can't flash the gen2 hardware. It is available on the release page [here](https://github.com/judge2005/EleksTubeIPS/releases/tag/v1.6.4). The instruction for uploading it manually are [here](https://github.com/judge2005/EleksTubeIPS/wiki/Installation). Read the installation instructions. The specific firmware you will need is [this one](https://github.com/judge2005/EleksTubeIPS/releases/download/v1.6.4/firmware-elekstubev2-1.6.4.bin).

If that is all too daunting I could add a v2 install page to the web site.

**Reply by IPSude** [reply]
Hmm yes the python stuff is unknown to me :) checking the v2 site you linked below.

**Comment by judge2005** [comment]
I added the v2 firmware to the web site [here](https://www.nixies.us/projects/elekstubeips-clock/elekstube-ips-v1-5/). Try that.

**Reply by IPSude** [reply]
Tried  this, appreciate the quick reply and effort :)
When I click on Full install nothing seems to happen, it prompts me to select the  corresponding COM port and then Error: Failed to execute 'open' on 'Serialport": The port is already open.

Same with Firmware only...

Configure WIfi gives:  An Error occured. Improv Wi-Fi Serial not detected.

**Reply by judge2005** [reply]
Well that just means that something else has the serial port open. Are you selecting the correct serial port? Do you have another application open that is talking to the clock?

**Reply by IPSude** [reply]
Finally managed to do the install, I am seeing the weather thingies now for Today and then rest unknown...
Any idea why it won't display the weather temperature for me?
![474158725_1279767096464001_1481535068713789687_n](https://github.com/user-attachments/assets/cdbedb02-f3e0-4e2d-a609-43471a1fa796)
![475674468_1298292491506581_4564075871884327221_n](https://github.com/user-attachments/assets/eb9a1e64-97a0-4183-849a-04d2ef8f2ad6)

**Reply by IPSude** [reply]
think I figured out the api key thing... it shows some cloud and temperature but the matrix screensaver pops up all the time. The weather is shows briefly one screen after the other and then the matrix.
The first screen doesn't show weather and in clock/weather mode the last remains blank with weather info...
![474441336_1029694955617749_196823068387237592_n](https://github.com/user-attachments/assets/8c941c27-e867-49be-a09c-11d10fbeb672)

**Reply by IPSude** [reply]
managed to get it to display the correct weather now :) thanks to the latitude longitude settings, don't know about the screensaver thing managed to keep the time or weather on screen now... but don't know what I did...
Dare not touch a button again now :p any chance to have the display alternate between time and weather every x min or something?
![474900081_1121223989665188_5311058519475334420_n](https://github.com/user-attachments/assets/49adfb3e-4f3c-40fc-aa88-0cc25521c7e9)

**Reply by judge2005** [reply]
Maybe - space is pretty tight in there and I'm a bit snowed under with other stuff ATM.

The matrix thing could be either the screensaver or the blanking time (where it goes to sleep for a certain period every 24 hours). You can force it into those modes with the buttons so it might have been reading as being pressed?

The wiki documents what the buttons (are supposed) to do.

They are handy because you can switch between displays using the buttons.

**Reply by IPSude** [reply]
Thank you for your time and explanations, much appreciated!

**Reply by IPSude** [reply]
One final question, can I still revert to the original firmware?

**Reply by IPSude** [reply]
Oh and I can confirm the webinterface doesn't work on apple devices only but also on android device with chrome webbrowser ;)

**Reply by judge2005** [reply]
> One final question, can I still revert to the original firmware?
> 
> 

Assuming you backed it up, yes.

**Reply by IPSude** [reply]
> > One final question, can I still revert to the original firmware?
> 
> Assuming you backed it up, yes.

Which I obviously forgot since I don't know python :p oh well the current firmware seems to be a working improvement, just need to figure out how to prepare new clock faces :)

**Reply by IPSude** [reply]
> I added the v2 firmware to the web site [here](https://www.nixies.us/projects/elekstubeips-clock/elekstube-ips-v1-5/). Try that.

Hey Judge, First of all awesome piece of custom firmware, thanks for that. Is there any chance of adding additional features?
I love the rainbow effect on the led lights for example but often forget to turn them off, any chance they could be turned off along with when the IPS leds are dimmed (same hours) for example?

**Reply by judge2005** [reply]
I think this is implemented in latest version https://github.com/judge2005/EleksTubeIPS/issues/38

**Reply by IPSude** [reply]
> I think this is implemented in latest version #38

Cheers! Thanks again for your great work and time and effort :)

Had to fiddle with my settings again to get everything as it was before (timezone and weather API etc...) was there a way to save those settings or you need  to reconfigure after each update?
Also how can I get notifications when new firmware or updates have been released?

**Reply by judge2005** [reply]
> > I think this is implemented in latest version #38
> 
> Cheers! Thanks again for your great work and time and effort :)
> 
> Had to fiddle with my settings again to get everything as it was before (timezone and weather API etc...) was there a way to save those settings or you need to reconfigure after each update? Also how can I get notifications when new firmware or updates have been released?

If I add a new setting, it will reset them all to the defaults. I normally say if that is the case though. Also, if you select ‘erase flash’ (or something like that), it will lose everything including the WiFi connection.

**Comment by IPSude** [comment]
Is there any risk of bricking my clock and do I need to back up the original firmware? If yes, how?
Thanks in advance.
Am new to this :)
